components/popover/index.js

```diff
@@ -133,17 +133,27 @@ class Popover extends Component {
 	}
 
 	setOffset() {
+		const { range } = this.props;
 		const { anchor, popover } = this.nodes;
 		if ( ! anchor || ! anchor.parentNode ) {
 			return;
 		}
 
-		const rect = anchor.parentNode.getBoundingClientRect();
 		const [ yAxis, xAxis ] = this.getPositions();
 		const isTop = 'top' === yAxis;
 		const isLeft = 'left' === xAxis;
 		const isRight = 'right' === xAxis;
-
+		let rect = anchor.parentNode.getBoundingClientRect();
```

Comment:
Seems wasteful to call `Element#getBoundingClientRect` only to have chance of overriding if `range` is specified. This looks like something that could go into the `else` of the below condition:

```js
let rect;
if ( range ) {
	// ...
} else {
	rect = anchor.parentNode.getBoundingClientRect();
}
```

Reply:
Done.